state-chain/runtime/src/runtime_apis.rs

```diff
@@ -344,6 +344,47 @@ pub struct TransactionScreeningEvents {
 	pub arb_events: Vec<BrokerRejectionEventFor<cf_chains::Arbitrum>>,
 }
 
+/// Struct that represents the BTC deposit addresses for a given private channel id.
+#[derive(Encode, Decode, TypeInfo, Serialize, Deserialize, Clone)]
+pub struct BtcDepositAddresses<AddressType> {
+	pub previous: AddressType,
+	pub current: AddressType,
+}
+
+#[derive(Encode, Decode, TypeInfo, Serialize, Deserialize, Clone)]
+pub struct VaultAddresses<AddressType> {
+	pub ethereum_vault: AddressType,
+	pub arbitrum_vault: AddressType,
+	pub bitcoin_vault: BtcDepositAddresses<AddressType>,
+	pub bitcoin_deposit_addresses: BTreeMap<AccountId32, BtcDepositAddresses<AddressType>>,
```

Comment:
As noted above, there is no `bitcoin_vault`, we only witness the broker private channel addresses. I don't think the deposit monitor cares about who owns them either, or which is previous/current (maybe @MxmUrw can confirm).

I think we can simplify to this:

```
struct VaultAddresses<AddressType> {
	pub ethereum: AddressType,
	pub arbitrum: AddressType,
	pub bitcoin: Vec<AddressType>,
}
```

(also, nit: we don't need the `Type` suffix. We already know it's a type.)

Reply:
Yep, the deposit monitor needs to know the owner in order to be able to filter the addresses according to it's configuration.